python/cuspatial/cuspatial/geometry/geoarrowbuffers.py

```diff
@@ -0,0 +1,640 @@
+    def offsets(self):
+        """
+        The offsets column of a geometry object contains the positions of
+        each sub-geometry. Each pair of values in the offsets column specifies
+        the beginning index of a sub-geometry in the `.xy` column and beginning
+        of the subsequent sub-geometry. Contains `n+1` values where `n` is the
+        number of sub-geometries. Starts always with 0.
+        """
+        return self._offsets
+
+    @offsets.setter
+    def offsets(self, offsets):
+        self._offsets = self._serialize(offsets)
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            rindex = index
+        else:
+            rindex = slice(index, index + 1, 1)
+        new_slice = slice(self.offsets[rindex.start], None)
+        if rindex.stop < len(self.offsets):
+            new_slice = slice(new_slice.start, self.offsets[rindex.stop])
+        result = self.xy[new_slice]
+        return result
+
+    def __repr__(self):
+        return (
+            f"{super().__repr__()}"
+            f"offsets:\n"
+            f"{self.offsets.__repr__()}\n"
+        )
+
+    def copy(self, deep=True):
+        if self.z is not None:
+            z = self.z.copy(deep)
+        else:
+            z = None
+        result = OffsetArray(self.xy.copy(deep), self.offsets.copy(deep), z)
+        return result
+
+    def __len__(self):
+        return len(self._offsets) - 1
+
+
+class LineArray(OffsetArray):
+    def __init__(self, xy, lines, mlines, z=None, data_locale=cudf):
+        """
+        A GeoArrow column of LineStrings. This format stores LineStrings and
+        MultiLineStrings from a single data source (Such as a GeoSeries).
+        Offset coordinates stored between pairs of mlines offsets specify
+        MultiLineStrings. Offset values that do not fall within a pair of
+        mlines are simple LineStrings.
+
+        Parameters
+        ---
+        xy : cudf.Series
+        lines : cudf.Series
+        mlines : cudf.Series
+        z : cudf.Series (optional)
+        """
+        super().__init__(xy, lines, z, data_locale=data_locale)
+        self.mlines = mlines
+
+    @property
+    def mlines(self):
+        """
+        The mlines column of a MultiLine object contains the positions of
+        each sub-geometry. Each pair of values in the mlines column specifies
+        the beginning index of a sub-geometry in the `.xy` column and beginning
+        of the subsequent sub-geometry. Contains `2n` values where `n` is the
+        number of sub-geometries.
+        """
+        return self._mlines
+
+    @mlines.setter
+    def mlines(self, mlines):
+        self._mlines = self._serialize(mlines)
+
+    def __repr__(self):
+        return (
+            f"{super().__repr__()}" f"mlines:\n" f"{self.mlines.__repr__()}\n"
+        )
+
+    def copy(self, deep=True):
+        base = super().copy(deep)
+        result = LineArray(base.xy, base.offsets, self.mlines.copy(), base.z,)
```

Comment:
Should `self.mlines.copy()` take in the deep parameter here?

Reply:
yes thanks